### PR TITLE
Use the address of symbolic conditional instructions

### PR DESCRIPTION
**Pull request by NeatMonster**
When two symbolic conditional instructions follow sequentially, the two paths of
the first instruction will be explored. But when the exploration reaches the
second instruction, because the at_symbolic_conditional flag has already been
set by the first one, only one of the two possible paths will be explored.

This commit fixes this issue by replacing the boolean flag by the address of the
instruction, which ensure the flag has been set for the current instruction.

---
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/1239)

### CODE REVIEW

**Comment by disconnect3d** [comment]
Can we add a test for it?

**Comment by NeatMonster** [comment]
Here is an unit test for it. It ensures that two branches would be explored each time.

File: tests/test_armv7cpu.py [hunk 1]
```diff
@@ -1695,3 +1696,35 @@ def test_arm_save_restore_cpu(self):
...
 
+    def test_symbolic_conditional(self):
+        asm = \
+        """
+          tst r0, r0
+          beq label
+          bne label
+        label:
+          nop
+        """
```
Comment (disconnect3d) [inline tests/test_armv7cpu.py]: Can we rewrite the code so the addresses are more explicit?

e.g.
```python
# in thumb mode each instruction has size of 2 bytes
# (the code starts at 0x1004)
asm = '\n'.join(
    '  tst r0, r0',  # 0x1004
    '  beq label'  # 0x1006
    '  bne label'  # 0x1008
    'label:'
    '  nop'  # 0x100a
)
```

Comment (NeatMonster) [reply tests/test_armv7cpu.py]: Done, see my latest commit. :smile:

**Comment by NeatMonster** [comment]
I've made the changes as requested, and added some more comments.

**Pull request merged**